app/assets/javascripts/games.js.coffee

```diff
@@ -1,12 +1,16 @@
-
 $ ->
   gameId = $('.chessboard').data('channelid')
 
   if (gameId != undefined)
     pusher = new Pusher '9d04d520abd8261569ea', { encrypted: true }
     channel = pusher.subscribe gameId
-    channel.bind 'refresh_event', (data)->
-      location.reload()
+    channel.bind 'move', (data)->
+      piece = $("[data-id=#{data.piece_id}]")
+
+      if piece.css('top') == '0px' && piece.css('left') == '0px'
+        animateMove(piece, data.destination)
+      else
+        movePieceElement(piece, data.destination)
```

Comment:
When the user drags a piece it updates the top and left offsets to move the piece. If they are both zero we can assume the piece has not been dragged and therefore we want to animate it into position. Essentially the second case is for the user that made the move and the other case is for everyone else. I'm definitely open to suggestions/patches to make this code clearer.   My JS tends to spaghettify rather quickly.